### PR TITLE
OOP school library: set up associations

### PR DESCRIPTION
**Pull request by okoyecharles**
## Description ✏️
In this project, I finished creating the remaining classes for our school library and created the associations between them.

## Changes made ♻️
- [x] Create Associations
- [x] Fix linter Errors

### CODE REVIEW

File: ruby/classroom.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Classroom
+  attr_accessor :label
+
```
Comment (MahmoudBakr23) [inline ruby/classroom.rb:4]: - [ ] For the associations between the `Classroom` and the `Student` classes to work properly when trying to invoke the `add_students` method it has to be set up `attr_reader :students` 🙏

File: ruby/book.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author
```
Comment (MahmoudBakr23) [inline ruby/book.rb]: - [ ] Please make sure to add the rentals to the associations in the `Book` class as well

File: ruby/rental.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+class Rental
+  attr_accessor :date
```
Comment (MahmoudBakr23) [inline ruby/rental.rb]: - [ ] For the same reason as in the request about the classroom and student classes, please add the person and book in the association with the rental class. 🙏

File: ruby/book.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+end
```
Comment (MahmoudBakr23) [inline ruby/book.rb:16]: - [ ] It seems that you forgot to create the custom method `add_rental` in the Book class to make a person be able to rent a book. Please do so and make sure to test it out in your terminal. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @OkoyeCharles 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Great work so far working with the associations between your classes
- Well done working with the custom methods in the classes
- Correct Linters
- Descriptive PR

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process improves the code you wrote, not criticizes it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

- [ ] Please fix the MIT link for a professional README file

Cheers and Happy coding!👏👨‍💻💻

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: ruby/book.rb (hunk 4 above)
Comment (okoyecharles) [reply ruby/book.rb:16]: Thanks very much for the review @MahmoudBakr23, I will implement that now.

**Review by rotimiazeez — APPROVED** [review]
# Hi @OkoyeCharles 👍🏽, 
## Good job implementing the required changes 👏🏽
### Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**